scripts/deploy.sh

```
@@ -0,0 +1,524 @@
+    # Check if environment file exists
+    local env_file="$PROJECT_ROOT/.env.$ENVIRONMENT"
+    if [[ ! -f "$env_file" ]]; then
+        warn "Environment file not found: $env_file"
+        if [[ -f "$PROJECT_ROOT/.env.${ENVIRONMENT}.template" ]]; then
+            info "Template found. Please copy and configure: cp .env.${ENVIRONMENT}.template .env.$ENVIRONMENT"
+        fi
+        error_exit "Environment configuration missing"
+    fi
+    
+    # Validate required environment variables
+    source "$env_file"
```

Shellcheck (0.10.0)

[warning] ShellCheck can't follow non-constant source. Use a directive to specify location.

(SC1090)